services/postgres/UserTopicService.js

```diff
@@ -9,29 +9,43 @@ class UserTopicService {
     this._userTopicHistoryModel = UserTopicHistoryModel;
     this.followTopic = this.followTopic.bind(this);
     this.getUserTopic = this.getUserTopic.bind(this);
+    this.getFollowTopicStatus = this.getFollowTopicStatus.bind(this);
   }
 
-  async getFollowTopic(user_id, topic_id) {
-    try {
-    } catch (error) {}
-  }
-
-  async followTopic(user_id, topic_id) {
+  async getFollowTopicStatus(user_id, topic_id) {
     try {
       let topicUser = await this.getUserTopic(user_id, topic_id);
       if (topicUser) {
-        await this.deleteUserTopic(user_id, topic_id);
         return true;
       } else {
-        await this.addUserTopic(user_id, topic_id);
         return false;
       }
+    } catch (error) {
+      console.log(error);
+      throw new InvariantError('error get follow topic status');
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getFollowTopicStatus` function is using a try-catch block to handle errors, which is good. However, it's logging the error and then throwing a new error with a generic message. This could make debugging difficult because the original error message, which might contain important information about what went wrong, is not included in the new error. I suggest including the original error message in the new error.

```diff
-       console.log(error);
-       throw new InvariantError('error get follow topic status');
+       throw new InvariantError(`error get follow topic status: ${error.message}`);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->